lib/library.rb

```diff
@@ -0,0 +1,76 @@
+require 'yaml'
+require 'date'
+require 'pry-byebug'
+
+class Library
+  attr_reader :collection, :return_date
+
+  def initialize
+    @collection = load_book_collection
+  end
+
+  def book_check_out(item, patron)
+    book = find_book(item)
+    case
+    when book == 'no book'
+      'Library does not have this book'
+    when book[:available] == false
+      'Book is not available. It will be returned ' +
+        (@collection[book[:index]][:return_date]).to_s
+    else
+      perform_transaction(book, patron)
+    end
+  end
+
+  def available_books
+    avail_books = []
```

Comment:
You'll also find a method on the `Enumerable` class that does just what you're trying to achieve here